helpers/deployer/steps/steps.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package steps
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"sort"
+	"strings"
+)
+
+const (
+	completedStatus = "COMPLETED"
+	failedStatus    = "FAILED"
+	pendingStatus   = "PENDING"
+)
+
+type Step struct {
+	Name   string `json:"name"`
+	Status string `json:"status"`
+	Error  string `json:"error"`
```

Comment:
nit: since we implement stringer on Step anyway, should we keep `Error  error` incase we want to do some error unwrapping in the future?

Reply:
I was not able to json marshal the `error` type.
I got a empty `{}` as the json representation.

Reply:
Gotcha, we may need to wrap the error and implement a MarshalJSON func. We can defer this until we need it.